### PR TITLE
[prometheus] Migrate to chart v2

### PR DESCRIPTION
**Pull request by zanhsieh**
Signed-off-by: zanhsieh <zanhsieh@gmail.com>

#### What this PR does / why we need it:
Initial attempt to move to chart v2, since helm v2 has been out of support from Nov. 15th, 2020.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus]`)

### CODE REVIEW

File: charts/prometheus/Chart.yaml [hunk 1]
```diff
@@ -23,4 +23,4 @@ maintainers:
   - name: naseemkullah
     email: naseem@transit.app
 engine: gotpl
-tillerVersion: ">=2.8.0"
+type: application
```
Comment (monotek) [inline charts/prometheus/Chart.yaml:26]: dependencies should go to chart.yaml too.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/README.md [hunk 2]
```diff
@@ -6,7 +6,8 @@ This chart bootstraps a [Prometheus](https://prometheus.io/) deployment on a [Ku
 
 ## Prerequisites
 
-- Kubernetes 1.3+ with Beta APIs enabled
+- Kubernetes 1.16+
```
Comment (monotek) [inline charts/prometheus/README.md:9]: If we go for kubernetes 1.16 all extension beta apis should be removed.

For example: https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/server/ingress.yaml#L10

**Review by monotek — CHANGES_REQUESTED** [review]
container images could be updated to.

**Comment by zanhsieh** [comment]
> container images could be updated to.

This one I will need to separate PR since too many moving parts.

**Review by monotek — CHANGES_REQUESTED** [review]
Imho it's perfect to do that if we introduce a new major version.

File: charts/prometheus/Chart.yaml [hunk 3]
```diff
@@ -23,4 +23,10 @@ maintainers:
...
+
```
Comment (monotek) [inline charts/prometheus/Chart.yaml:27]: please remove empty line

**Review by monotek — CHANGES_REQUESTED** [review]
please remove empty line in chart.yaml

**Comment by zanhsieh** [comment]
> please remove empty line in chart.yaml

@monotek 
The lint will complain about this. It needs an empty line on the end.

**Review by monotek — APPROVED** [review]

**Pull request merged**